apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/sharing-modal/index.tsx

```diff
@@ -0,0 +1,274 @@
+import { store as noticesStore } from '@wordpress/notices';
+import classnames from 'classnames';
+import React from 'react';
+import ClipboardButton from 'calypso/components/forms/clipboard-button';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormLabel from 'calypso/components/forms/form-label';
+import SocialLogo from 'calypso/components/social-logo';
+import { useShouldShowFirstPostPublishedModal } from '../../../dotcom-fse/lib/first-post-published-modal/should-show-first-post-published-modal-context';
+import useShouldShowSellerCelebrationModal from '../../../dotcom-fse/lib/seller-celebration-modal/use-should-show-seller-celebration-modal';
+import useShouldShowVideoCelebrationModal from '../../../dotcom-fse/lib/video-celebration-modal/use-should-show-video-celebration-modal';
+import postPublishedImage from './images/illo-share.svg';
+import useSharingModalDismissed from './use-sharing-modal-dismissed';
+import './style.scss';
+
+type CoreEditorPlaceholder = {
+	getCurrentPost: ( ...args: unknown[] ) => { link: string; title: string };
+	getCurrentPostType: ( ...args: unknown[] ) => string;
+	isCurrentPostPublished: ( ...args: unknown[] ) => boolean;
+};
+const FB_APP_ID = '249643311490';
+
+const SharingModal: React.FC = () => {
+	const isDismissedDefault = window?.sharingModalOptions?.isDismissed || false;
+	const { launchpadScreenOption } = window?.launchpadOptions || {};
+	const { isDismissed, updateIsDismissed } = useSharingModalDismissed( isDismissedDefault );
+
+	const { link, title } = useSelect(
+		( select ) => ( select( 'core/editor' ) as CoreEditorPlaceholder ).getCurrentPost(),
+		[]
+	);
+	const postType = useSelect(
+		( select ) => ( select( 'core/editor' ) as CoreEditorPlaceholder ).getCurrentPostType(),
+		[]
+	);
+
+	const isCurrentPostPublished = useSelect(
+		( select ) => ( select( 'core/editor' ) as CoreEditorPlaceholder ).isCurrentPostPublished(),
+		[]
+	);
+	const previousIsCurrentPostPublished = useRef( isCurrentPostPublished );
+	const shouldShowFirstPostPublishedModal = useShouldShowFirstPostPublishedModal();
+	const shouldShowSellerCelebrationModal = useShouldShowSellerCelebrationModal();
+	const shouldShowVideoCelebrationModal =
+		useShouldShowVideoCelebrationModal( isCurrentPostPublished );
+
+	const [ isOpen, setIsOpen ] = useState( false );
+	const closeModal = () => setIsOpen( false );
+	const { createNotice } = useDispatch( noticesStore );
+
+	const siteSlug = window.location.hostname;
+	const subscribersUrl = `https://wordpress.com/people/subscribers/${ siteSlug }`;
+
+	useEffect( () => {
+		// The first post will show a different modal.
+		if (
+			! shouldShowFirstPostPublishedModal &&
+			! shouldShowSellerCelebrationModal &&
+			! shouldShowVideoCelebrationModal &&
+			launchpadScreenOption !== 'full' &&
+			! previousIsCurrentPostPublished.current &&
+			isCurrentPostPublished &&
+			postType === 'post'
+		) {
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg)

It might be bug-prone to manage this kind of condition every time we add a new modal. (A dev who will add a new modal needs to know this condition exists somehow.)🙄
Maybe we should add a shared context (or whatever), such as `hasPostEditorModal`, or create something like a "mediator" that controls the display of a modal? 
What do you think?

Reply:
hmmm I'm not sure, I think the other modals have managed so far because they all show in diferent contexts. But if two modals where to show at the same time, we need to decide which has priority. But then the problem is that every modal needs to know about every other modal it may conflict with.

I guess it depends on how many more modals we plan on adding, if only one or two more, maybe this current approach is ok, but any more and it becomes increasingly complex.

It's also complicated because of the direction to move new features into jetpack. Modals implemented there won't have access to these flags :/ 

My preference is to go with this approach for now and then maybe next time we add a modal, we should migrate all modals into jetpack, and think about if we need to change the approach then.